### PR TITLE
docs: update release docs to prevent orphaned merge commits

### PR DESCRIPTION
**Pull request by booc0mtaco**
### Summary:

make use of the updated repository config to release without creating unsynced merge commits

### Test Plan:

- n/a

### CODE REVIEW

File: docs/PUBLISHING.md [hunk 1]
```diff
@@ -65,18 +65,18 @@ Merge the PR through a **merge commit**:
...
-    - #eng-n00bs in the CZI Education workspace
-    - #proj-edu-design-system in the CZI Education workspace
-11. Update the package in the main apps that use it (for major versions):
-    - [cra-template-edu](https://github.com/chanzuckerberg/frontend-libs/tree/main/packages/cra-template-edu) - in the template.json and package.json
-    - [edu-stack](https://github.com/chanzuckerberg/edu-stack) - in the package.json
+7. Pull down the most up-to-date version of main: `git checkout main && git pull && yarn build`
+8. Publish the package: `npm publish`
+9. Create a [new release](https://github.com/chanzuckerberg/edu-design-system/releases) based on the new tag. Use the same text used for the pull request description above (from CHANGELOG.md). This will automatically post to [relevant slack channels](https://slack.github.com/):
+10. Lastly, run the following to "back merge" release changes to `next`:
+    - git checkout main && git pull origin main && git checkout next && git merge main && git push
```
Comment (jinlee93) [inline docs/PUBLISHING.md:74]: with `git merge main` that would be recursive strategy, would that be fine?

**Comment by booc0mtaco** [comment]
We'll be discussing further simplifications next week, but these steps allow for cleaner releases in between, preventing orphaned merge commits

File: docs/PUBLISHING.md (hunk 1 above)
Comment (booc0mtaco) [reply docs/PUBLISHING.md:74]: I believe so. At this point, `main` will have a merge commit with one additional commit in for the release information. We want to merge that commit AND merge commit back to `next`. if we create an additional merge commit on `next`, it's OK in this case, as that will only push `next` up by one additional "commit" with no other changes attached.

**Review by jinlee93 — APPROVED** [review]
Thanks!

**Pull request merged**